syntax = "proto3";
package auth;

// Serviço de autenticação
service AuthService {
  // Login retorna token JWT
  rpc Login(LoginRequest) returns (LoginResponse);

  // Retorna a chave pública no formato JWK
  rpc GetJWKS(Empty) returns (JWKSResponse);
}

// Requisição de login
message LoginRequest {
  string username = 1;
  string password = 2;
}

// Resposta de login
message LoginResponse {
  string access_token = 1;
  string token_type = 2;
}

// Requisição vazia
message Empty {}

// Resposta do JWKS
message JWKSResponse {
  string keys_json = 1;
}
